embedded-graphics/src/primitives/polyline/thick_points.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    prelude::Point,
+    primitives::{
+        line::Side,
+        line_joint::JointKind,
+        polyline::triangle_iterator::TriangleIterator,
+        triangle::{Points, Triangle},
+        ContainsPoint, Primitive,
+    },
+    style::StrokeAlignment,
+};
+
+/// Remembers the previous 5 points to avoid overdraw.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+struct LookbackBuffer {
+    points: [Point; 5],
+    count: u8,
+}
+
+impl LookbackBuffer {
+    /// Creates a new buffer.
+    pub const fn new() -> Self {
+        Self {
+            points: [Point::zero(); 5],
+            count: 0,
+        }
+    }
+
+    /// Shifts a point to the front of the buffer.
+    pub fn add(&mut self, p: Point) {
+        self.points[4] = self.points[3];
+        self.points[3] = self.points[2];
+        self.points[2] = self.points[1];
+        self.points[1] = self.points[0];
+        self.points[0] = p;
+        if self.count < 5 {
+            self.count += 1;
+        }
+    }
+
+    /// Returns whether the given point is inside the newest tracked triangle.
+    pub fn prev1_contains(&self, p: Point) -> bool {
+        self.count >= 3 && Triangle::new(self.points[0], self.points[1], self.points[2]).contains(p)
```

Comment:
This is the most painful check. Any simplification could yield significant perf gains.